app/utils/rentry/wrapper.py

```diff
@@ -108,100 +140,150 @@ def new(self, text: str) -> Any:
 
         except Exception as e:
-            logger.error(
-                f"An error occurred while fetching rentry.co content: {str(e)}"
-            )
-            show_fatal_error(
-                title="Error",
-                text=f"An error occurred: {str(e)}",
-            )
+            RentryError().show_error_warning(e)  # Handle exceptions during the process
+
+
+class RentryError:
+    """Class to handle errors related to Rentry operations."""
+
+    def handle_upload_failure(self, response: dict[str, Any]) -> None:
+        """
+        Log and handle upload failure details.
+
+        Args:
+            response (dict[str, Any]): A dictionary containing the response details from the upload attempt.
+        """
+        error_content = response.get("content", "Unknown")  # Extract error content
+        errors = [
+            error.strip()
+            for error in response.get("errors", "").split(".")
+            if error.strip()
+        ]
+
+        logger.error(
+            f"Rentry upload process failed with Error: {error_content}"
+        )  # Log main error
+
+        if errors:
+            for error in errors:
+                logger.error(f"Detail: {error}")  # Log individual errors
+        else:
+            logger.error("No specific error details available.")
+
+        logger.error(
+            f"Rentry.co upload failed! Error: {error_content}. Details: {', '.join(errors) if errors else 'None'}"
+        )
+
+    def show_info(self, response: requests.Response) -> None:
+        """
+        Show information about a failed fetch attempt.
+
+        Args:
+            response (requests.Response): The response object containing status code and reason.
+        """
+        logger.warning(
+            f"Rentry returned status code: {response.status_code}. Reason: {response.reason}"
+        )
+        InformationBox(
+            title="Failed to fetch Rentry Content",
+            text=f"Rentry returned status code: {response.status_code}",
+            information="RimSort failed to fetch the content from the provided Rentry link. This may be due to an invalid link, your internet connection, or Rentry.co being down. It may also be the result of a captcha. Please try again later.",
+            details=response.reason,
+            icon=QMessageBox.Icon.Warning,
+        ).exec()
+
+    def show_error_warning(self, e: Exception) -> None:
+        """
+        Show a warning for network errors.
+
+        Args:
+            e (Exception): The exception that occurred during the network operation.
+        """
+        logger.error(f"A network error occurred while processing Rentry: {str(e)}")
+        show_fatal_error(
+            title="Network Error",
+            text="A network error occurred while processing Rentry, Please check if your internet is working",
+            details=f"{str(e)}",
+        )
+        return None  # Return None to indicate failure
```

Comment:
Change this to handle exceptions in a generic way. It might be due to network error, or something completely else. Just use generic language saying error/exception occurred, upload your logs to discord/github  etc.

Reply:
Maybe rename the method to something simpler too. Since ```show_error_warning``` is unclear whether its an error or a warning, especially with the comment saying its a warning. Maybe a generic ```show_error``` or ```show_generic_error```?

Reply:
changes done